src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeFetchers.scala

```diff
@@ -155,24 +155,25 @@ trait BlocksDataFetchers {
     import JsonDecoders.Circe.Rights._
 
     /** the input type, e.g. ids of data */
-    override type In = BlockHash
+    override type In = (Option[Int], Option[Int], BlockHash)
 
     /** the output type, e.g. the decoded block data */
     override type Out = List[BakingRights]
 
     /** the encoded representation type used e.g. some Json representation */
     override type Encoded = String
 
-    private val makeUrl = (hash: BlockHash) => s"blocks/${hash.value}/helpers/baking_rights"
+    private val makeUrl = (blockData: (Option[Int], Option[Int], BlockHash)) => s"blocks/${blockData._3.value}/helpers/baking_rights"
 
     /** an effectful function from a collection of inputs `T[In]`
       * to the collection of encoded values, tupled with the corresponding input `T[(In, Encoded)]`
       */
-    override val fetchData: Kleisli[Future, List[BlockHash], List[(BlockHash, String)]] =
+    override val fetchData: Kleisli[Future, List[(Option[Int], Option[Int], BlockHash)], List[((Option[Int], Option[Int], BlockHash), String)]] =
```

Comment:
If you don't want to get mad with heavy signatures, you might want to use the alias `In` for the tuple, defined a little above

Reply:
Sure.